creator/ingest_runs/data_generator/study_generator.py

```diff
@@ -435,48 +467,102 @@ def workflow_type(row):
-            "filepath": "Filepath",
-            "source_read": "Source Read",
+            "harmonized_path": "Filepath",
+            "source_path": "Source Read",
         }
         df = df[
             list(col_map.keys()) +
             ["Data Type", "Workflow Type", "Cavatica ID", "Cavatica Task ID"]
         ].rename(columns=col_map)
 
-        df.to_csv(fp, sep="\t", index=False)
-        self.logger.info(f"Wrote {fp}")
-
         return df
 
-    def _assert_load(self, endpoint, kf_ids):
+    def _assert_load(self):
         """
-        Ensure all kf ids exist and entity count in Data Service
-        matches len(kf_ids)
+        Ensure all kf ids in the data files got loaded into Data Service
         """
-        self.logger.info(f"Check all {endpoint} were loaded")
-
-        base_url = f"{DATASERVICE_URL}/{endpoint}"
-        for kfid in kf_ids:
+        # Extract kf ids
+        bio_df = self.dataframes["bio_manifest.tsv"]
+        gf_df = self.dataframes["sequencing_manifest.tsv"]
+        kfids = {
+            "families": bio_df[["kf_id_family"]],
+            "participants": bio_df[["kf_id_participant"]],
+            "biospecimens": bio_df[["kf_id_biospecimen"]],
+            "genomic-files": gf_df[["kf_id_genomic_file"]],
+        }
+        for endpoint, df in kfids.items():
+            self.logger.info(f"Check all {endpoint} were loaded")
             # Check kf ids
-            url = f"{base_url}/{kfid}"
-            self.logger.info(f"Checking {url}")
-            resp = requests.get(url)
-            resp.raise_for_status()
-
-        # Check exact count
-        count = len(kf_ids)
-        if endpoint == "genomic-files":
-            params = {"harmonized": False}
-        else:
-            params = None
-        resp = requests.get(base_url, params=params)
-        total = resp.json()["total"]
-        assert total == count, (
-            f"{endpoint} expected count {count} != {total} found"
-        )
+            kfids = df[df.columns[0]].drop_duplicates().values.tolist()
+            base_url = f"{self.dataservice_url}/{endpoint}"
+            for kfid in kfids:
+                url = f"{base_url}/{kfid}"
+                self.logger.info(f"Checking {url}")
+                resp = requests.get(url)
+                resp.raise_for_status()
 
+            # Check exact count
+            params = {"study_id": self.study_id}
+            count = len(kfids)
+            if endpoint == "genomic-files":
+                params["harmonized"] = False
+            resp = requests.get(base_url, params=params)
+            total = resp.json()["total"]
+            assert total == count, (
+                f"{endpoint} expected count {count} != {total} found"
+            )
 
-if __name__ == '__main__':
-    sg = StudyGenerator()
-    sg.ingest_study(initialize=False, create_files=False)
-    # sg.generate_files(create=True)
+
+def delete_entities(dataservice_url, study_id=None, logger=None):
```

Comment:
Should probably be part of the `kf-utils-python` repo?

Reply:
Unfortunately kf-utils-python doesn't have any testing setup so we'll probably have to leave this here for now